### DISCUSSION TITLE
How to have uppercase on a TextInput

### DISCUSSION THREAD

**Discussion by eaziweb95** [filamentphp/filament#2724] [Help] [answered]
How to have uppercase on a TextInput to force in Maj the input characters
Thanks

**Comment by danharrin** [comment, marked as answer]
I don't think HTML inputs have an option to do this

**Reply by eaziweb95** [reply]
Is it possible to add the javascript function in the HTML input ?
onChange="this.value=this.value.toUpperCase();"

**Reply by danharrin** [reply]
`->extraInputAttributes(['onChange' => 'this.value = this.value.toUpperCase()'])`

**Reply by eaziweb95** [reply]
Fine ! It works ! Thanks a lot !

**Reply by eaziweb95** [reply]
is there another method because finally it doesnt work well

**Comment by wdog** [comment]
alternative

```php
TextInput::make('name')
    .....
    ->dehydrateStateUsing(fn(string $state): string => Str::upper($state))
```
